cmd/sovereignnode/chainSimulator/sovereignChainSimulator.go

```diff
@@ -63,9 +67,11 @@ func NewSovereignChainSimulator(args ArgsSovereignChainSimulator) (chainSimulato
 		return rating.NewSovereignRatingsData(arg)
 	}
 	args.CreateIncomingHeaderSubscriber = func(config *config.NotifierConfig, dataPool dataRetriever.PoolsHolder, mainChainNotarizationStartRound uint64, runTypeComponents factory.RunTypeComponentsHolder) (process.IncomingHeaderSubscriber, error) {
+		config = &alteredSovConfigs.NotifierConfig
 		return incomingHeader.CreateIncomingHeaderProcessor(config, dataPool, mainChainNotarizationStartRound, runTypeComponents)
 	}
 	args.CreateRunTypeComponents = func(argsRunType runType.ArgsRunTypeComponents) (factory.RunTypeComponentsHolder, error) {
+		*configs.SovereignExtraConfig = alteredSovConfigs
```

Comment:
I don't think that is really necessary to change the pointer here, you can just put `alteredSovConfigs` below in the return.
Or, If you need to have the `configs.SovereignExtraConfig` changed, just updated it at line 54 with `configs.SovereignExtraConfig = &cfg.GeneralConfig.SovereignConfig` and deleted the alteredSovConfigs var

Reply:
+1

Reply:
See above comment, as for extra variable, deleted it and passed directly

Reply:
The comment above is not valid. The `Or` is still an alternative if you want.